### PR TITLE
refactor: extract interface for Database Config

### PR DESCRIPTION
**Pull request by anugrahsinghal**
now each DB config can have a defined set of params it needs to offer
This will be useful for config of multiple DB that
the product needs to support and
different db configs might have different validations in place

### CODE REVIEW

File: src/main/java/io/supertokens/storage/sql/config/PostgreSQLConfig.java [hunk 1]
```diff
@@ -320,4 +345,19 @@ void validateAndInitialise() {
...
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return "org.postgresql.Driver";
+    }
+
+    @Override
+    public String getTableNamePrefix() {
+        return "";
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/sql/config/PostgreSQLConfig.java:360]: This should not be empty.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Comment by rishabhpoddar** [comment]
Also please see some of my comments in the other PR. That might affect this one.

File: src/main/java/io/supertokens/storage/sql/config/PostgreSQLConfig.java (hunk 1 above)
Comment (anugrahsinghal) [reply src/main/java/io/supertokens/storage/sql/config/PostgreSQLConfig.java]: fix added

**Comment by rishabhpoddar** [comment]
hey @anugrahsinghal i am closing this PR since this has duplicate code from https://github.com/supertokens/supertokens-sql-plugin/pull/14/files

**Comment by anugrahsinghal** [comment]
yeah makes sense

did the same changes in the other PR too

**Pull request closed without merging**